presto-main/src/main/java/io/prestosql/sql/analyzer/StatementAnalyzer.java

```diff
@@ -326,12 +331,14 @@ public Scope analyzeForUpdate(Table table, Optional<Scope> outerQueryScope)
         private final Optional<Scope> outerQueryScope;
         private final WarningCollector warningCollector;
         private final boolean isUpdateQuery;
+        private final UpdateKind updateKind;
 
-        private Visitor(Optional<Scope> outerQueryScope, WarningCollector warningCollector, boolean isUpdateQuery)
+        private Visitor(Optional<Scope> outerQueryScope, WarningCollector warningCollector, boolean isUpdateQuery, UpdateKind updateKind)
```

Comment:
`isUpdateQuery` and `updateKind` should be folded into a single argument. Otherwise, this allows for non-sensical combinations such as `isUpdateQuery == false` and `updateKind == UPDATE`.

Reply:
Good point - - as you suggest below, conveying both intentions in an `Optional<UpdateKind>` is the right thing, and removes the cursed `NO_UPDATE` enum value.  Done!